File: JULY-2025/05-07-2025/p1394.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findLucky = function(arr) {
     let obj ={}
    let answer=0
 for(let i =0;i<arr.length;i++){
      obj[arr[i]] = obj[arr[i]]? obj[arr[i]]+1:1
        
 }
 
 for(let j in obj){
   if(j == obj[j]){
       if(answer<j){
        answer = obj[j]
       }
   }
 }
return answer ===0?-1:answer
};

File: JULY-2025/05-07-2025/p1394.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int findLucky(int[] arr) {
        Map<Integer, Integer> map = new HashMap();
        for (int i : arr) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        int max = -1;
        System.out.println(map);
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() == entry.getKey()) {
                max = Math.max(max, entry.getValue());
            }
        }
        return max;
    }
}